### PR TITLE
mail/postfix: Add support for header_checks

### LINKED ISSUE #1570: postfix: header_checks for black / whitelisting
**Is your feature request related to a problem? Please describe.**
At the moment it is not possible to filter incoming mail (black / whitelisting), when mail is coming from a forwarder (SRS0=...).
Because the check_sender_access looks in MAIL FROM in envelope.
In a forwarded mail (SRS), the MAIL FROM is not the same as the "from" header. 
So there will never be a match.

**Describe the solution you'd like**
a solution could be to use header checks.
Here it is possible to check the "from" header with regex.
And this is just a small use case, with header checks there are much more possibilitys of filtering.

http://www.postfix.org/header_checks.5.html

### PR DESCRIPTION
**Pull request by Starkstromkonsument**
See http://www.postfix.org/header_checks.5.html

closes #1570

### CODE REVIEW

File: mail/postfix/src/opnsense/mvc/app/models/OPNsense/Postfix/Headerchecks.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                    <Required>Y</Required>
+                </enabled>
+                <expression type="TextField">
+                    <Required>Y</Required>
+                </expression>
+                <filter type="OptionField">
+                    <Required>Y</Required>
+                    <OptionValues>
+                        <WHILE_DELIVERING>WHILE DELIVERING MAIL</WHILE_DELIVERING>
+                        <WHILE_RECEIVING>WHILE RECEIVING MAIL</WHILE_RECEIVING>
```
Comment (fabianfrz) [inline mail/postfix/src/opnsense/mvc/app/models/OPNsense/Postfix/Headerchecks.xml:19]: I would not write that in uppercase

Comment (Starkstromkonsument) [reply mail/postfix/src/opnsense/mvc/app/models/OPNsense/Postfix/Headerchecks.xml:19]: Okay, I'll submit a commit changing this within the next few days. Do you mean the tag or the text or both?

Comment (fabianfrz) [reply mail/postfix/src/opnsense/mvc/app/models/OPNsense/Postfix/Headerchecks.xml:19]: I only care for the display text - if possible it should be as translatable as possible.

File: mail/postfix/Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 PLUGIN_NAME=		postfix
-PLUGIN_VERSION=		1.14
+PLUGIN_VERSION=		1.15
 PLUGIN_REVISION=	1
```
Comment (mimugmail) [inline mail/postfix/Makefile:3]: Can you remove the REVISION line please

File: mail/postfix/src/opnsense/mvc/app/controllers/OPNsense/Postfix/forms/dialogEditPostfixHeadercheck.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        <id>headercheck.enabled</id>
+        <label>Enabled</label>
+        <type>checkbox</type>
+        <help>This will enable or disable the header_check rule.</help>
+    </field>
+    <field>
+        <id>headercheck.expression</id>
+        <label>Expression</label>
+        <type>text</type>
+        <help><![CDATA[Set a regexp (POSIX regular expression) and an action to process like <pre>/^\s*User-Agent/ IGNORE</pre><b>Note: The regexp is not validated by this form. Please test it carefully.</b>]]></help>
```
Comment (mimugmail) [inline mail/postfix/src/opnsense/mvc/app/controllers/OPNsense/Postfix/forms/dialogEditPostfixHeadercheck.xml:12]: Can you remove bolding? normal test, pre, then bolding .. looks bit strange.

**Comment by Starkstromkonsument** [comment]
Thank you for your reviews. Do I have to do anything else for this PR to be merged? I'm sorry for asking, it's my first one for this project.

**Comment by mimugmail** [comment]
@fichtner will have a final look, but bit busy the last weeks with 20.7 release

**Comment by mimugmail** [comment]
@fichtner I already approved, if it's OK for you to allow regex in a text field this one is ready for merge :)

**Comment by fichtner** [comment]
Merged, thanks!

@Starkstromkonsument as a heads-up we will strip the GitHub link since nobody will be able to contact you from the link to GitHub profile directly. I'm happy to switch this for an e-mail address (and real name if you are up for it).

**Pull request merged**